project-koku/koku
# Support filter and group_by AWS product_family

AWS costs supports filter and group_by on product_family to support AWS UI enhancements

**Testing**
1.  Add cost-usage-bucket AWS provider
2.  Costs API group_by product_family wildcard
3.  Costs API group_by product_family filter on Compute
4.  Costs API group_by product_family filter on Storage
5.  Costs API filter product_family filter on Storage
6.  Costs API filter product_family filter on Compute
7. Costs API filter on invalid product_family. Verify 0 data returned.  NO error
8. Verify that /api/v1/reports/aws/storage/?filter[product_family]=Compute returns no results
9.  Verify that /api/v1/reports/aws/storage/?filter[product_family]=Storage returns same results as /api/v1/reports/aws/storage/?filter[product_family]=* and /api/v1/reports/aws/storage/
10. Execute order_by product_family

**Test Results**
[koku_issue_684_ut.txt](https://github.com/project-koku/koku/files/2933258/koku_issue_684_ut.txt)


Addresses #684
